### PR TITLE
feat: serializeJson reference and coercion to parseJson

### PR DESCRIPTION
**Pull request by odyslam**

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
🔥ty!

File: src/cheatcodes/serialize-json.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+### Example
+
+Let's assume we want to write the following JSON to a file:
+
+{ "boolean": true, "number": 342, "object": { "title": "finally json serialization" } }
+
+```solidity
+string memory json1 = "some key";
+vm.serializeBool(json1, "boolean", true);
+vm.serializeBool(json1, "number", uint256(342));
```
Comment (mds1) [inline src/cheatcodes/serialize-json.md:99]:
```suggestion
vm.serializeUint(json1, "number", uint256(342));
```

File: src/cheatcodes/parse-json.md [hunk 2]
```diff
@@ -35,6 +37,23 @@ To read more about the syntax, you can visit the [README](https://crates.io/crat
 - An array is encoded as a dynamic array of the type of it's first element
 - An object (`{}`) is encoded as a `tuple`
 
+### Type Coercion
+
+As described above, parseJSON needs to deduce the type of JSON value and that has some inherent limitations. For that reason, there is a sub-family of `parseJson*` cheatcodes that coerce the type of the returned value.
+
+For example `vm.parseJsonUint(json, key)` will coerce the value to a `uint256`. That means that it can parse all the following values and return them as a `uint256`. That includes a number as type `number`, a stringified number as a `string` and of course it's hex representation.
+
+````js
```
Comment (mds1) [inline src/cheatcodes/parse-json.md:46]: removes a backtick + changes language
```suggestion
```json
```

File: src/cheatcodes/parse-json.md [hunk 3]
```diff
@@ -48,7 +67,7 @@ The following JSON
     a: 43,
     b: "sigma"
 }
-```
+````
```
Comment (mds1) [inline src/cheatcodes/parse-json.md:70]: extra backtick here too, not sure if it affects rendering

**Pull request merged**